trieste/utils/pareto.py

```diff
@@ -0,0 +1,309 @@
+from typing import Mapping
+
+import gpflow
+import numpy as np
+import tensorflow as tf
+
+from trieste.data import Dataset
+from trieste.type import TensorType
+
+jitter = gpflow.config.default_jitter()
+
+
+def atleast2d(tensor: tf.Tensor) -> tf.Tensor:
+    if tf.rank(tensor) <= 1:
+        return tf.expand_dims(tensor, axis=0)
+    return tensor
```

Comment:
what's the use case for this? it's much easier to reason about functions that take just one shape and return one shape (modulo broadcasting dimensions)